rust-src/concordium_base/src/random_oracle/mod.rs

```diff
@@ -1,6 +1,8 @@
 //! This module provides the random oracle replacement function needed in the
 //! sigma protocols, bulletproofs, and any other constructions. It is based on
-//! SHA3.
+//! SHA3. It is used in non-interactive proofs and plays the same role as a random
+//! oracle would play in the corresponding interactive protocol (via
+//! Fiat–Shamir transformation).
```

Comment:
Would it be good to mention that our implementation is based on ideas coming from the `merlin transcript` as you discovered.